CSharpMinifier/Rewriters/CSharpRewriter.cs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        private SyntaxNode RenameAll(SyntaxNode node)
+        {
+            var mscorlib = MetadataReference.CreateFromFile(typeof(object).Assembly.Location);
+            _workspace = new AdhocWorkspace();
+            var project = _workspace.AddProject("MinifierProject", LanguageNames.CSharp);
+            project = project.AddMetadataReference(mscorlib);
+            _workspace.TryApplyChanges(project.Solution);
+            var source = node.GetText();
+            var document = _workspace.AddDocument(project.Id, "Doc", source);
+            _semanticModel = document.GetSemanticModelAsync().Result;
+            var newNode = _semanticModel.SyntaxTree.GetRoot();
+            foreach (KeyValuePair<VariableDeclaratorSyntax, string> variable in _identifierGenerator.RenamedVariables)
+            {
+                var nodeToSearch = newNode.DescendantNodes()
+                    .OfType<VariableDeclaratorSyntax>().FirstOrDefault(x => x.Identifier.ValueText.Equals(variable.Key.Identifier.ValueText));
+                (newNode, document) = Rename(nodeToSearch, document, variable.Value);
+            }
+            foreach (var classToRename in _identifierGenerator.RenamedTypes)
+            {
+                var nodeToSearch = newNode.DescendantNodes()
+                    .OfType<ClassDeclarationSyntax>().FirstOrDefault(x => x.Identifier.ValueText.Equals(classToRename.Key.Identifier.ValueText));
+                (newNode, document) = Rename(nodeToSearch, document, classToRename.Value);
+            }
+            foreach (var propertyToRename in _identifierGenerator.RenamedProperties)
+            {
+                var nodeToSearch = newNode.DescendantNodes()
+                    .OfType<PropertyDeclarationSyntax>().FirstOrDefault(x => x.Identifier.ValueText.Equals(propertyToRename.Key.Identifier.ValueText));
+                (newNode, document) = Rename(nodeToSearch, document, propertyToRename.Value);
+            }
+
+            return newNode;
+        }
+
+        private (SyntaxNode node, Document document) Rename(SyntaxNode nodeToRename, Document document, string newName)
+        {
+            var symbolInfo = _semanticModel.GetSymbolInfo(nodeToRename).Symbol ?? _semanticModel.GetDeclaredSymbol(nodeToRename);
+            var solution = Renamer.RenameSymbolAsync(document.Project.Solution, symbolInfo, newName,
+                _workspace.Options).Result;
+            _workspace.TryApplyChanges(solution);
+            document = _workspace.CurrentSolution.GetDocument(document.Id);
+            _semanticModel = document.GetSemanticModelAsync().Result;
+            return (_semanticModel.SyntaxTree.GetRoot(), document);
+        }
+
+        public override SyntaxNode VisitLocalDeclarationStatement(LocalDeclarationStatementSyntax node)
+        {
+            if (_options.LocalVarsCompressing)
+            {
+                _identifierGenerator.GetNextName(node.Declaration.Variables.First());
```

Comment:
I don't like the idea of method mutability that starts with `Get`. Try to convert this method to immutable or use another name.